### PR TITLE
fix: ncyBreadcrumbIgnore flag

### PR DESCRIPTION
**Pull request by kazinov**
Provides the way to ignore some views when rendering breadcrumbs. To achieve this the property ```ncyBreadcrumbIgnore``` should be set to true on the view controller scope:

```javascript
function ViewController($scope) {
       $scope.ncyBreadcrumbIgnore = true;
}
```
This allows to deal with multiple views related to the same state. 
Closes the issue #42 and possibly the issue #62 
[This plunker](http://plnkr.co/edit/yeODlzws6gv05tZ7nURd?p=preview) demonstrates a common problem with footer view which could be solved with this fix.

### CODE REVIEW

**Comment by ncuillery** [comment]
Hi, many thanks for the PR.

The idea of attach a configuration param to the scope sounds a bit dirty to me (I prefer centralize the breadcrumb configuration in the same place) but I recognize it's a simple and effective way to solve two major issues. 

It LGTM. I'll make a detailled review of this.

File: test/mock/test-modules.js [hunk 1]
```diff
@@ -117,3 +117,35 @@ angular.module('ncy-sample-conf', ['ncy-sample', 'ngMock']).config(function($url
     $httpBackend.when('GET', 'views/room_detail.html').respond('dummy room_detail view');
     $httpBackend.when('GET', 'views/room_form.html').respond('dummy room_form view');
 });
+
+/**
+ * Module with abstract state with footer view
+ */
+angular.module('ncy-abstract-interpolation-conf', []).config(function($stateProvider) {
```
Comment (ncuillery) [inline test/mock/test-modules.js]: I think the problem this PR solve is more related to multiple views than abstract view.

Tell me if I'm wrong but it doesn't matter if the state is abstract or not (I remove the `abstract: true` line in the plunker and the problem still occurs).

I suggest naming the conf `'ncy-multiple-interpolation-conf'`here (and update the describe's name below).

File: test/spec/directive-ncyBreadcrumbIgnore-test.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            $controllerProvider.register('ACtrl', function ($scope) {
+                $scope.ncyBreadcrumbIgnore = true;
+            });
+        });
+    });
+
+    describe('when ncyBreadcrumbIgnore property equals true on parent view scope', function () {
+        describe('when the parent view located before child view ', function () {
+            beforeEach(inject(function ($rootScope, $compile, $controller) {
+                controller = $controller;
```
Comment (ncuillery) [inline test/spec/directive-ncyBreadcrumbIgnore-test.js]: As far as I can see, this variable is unused ;-)

File: test/mock/test-modules.js [hunk 3]
```diff
@@ -117,3 +117,35 @@ angular.module('ncy-sample-conf', ['ncy-sample', 'ngMock']).config(function($url
...
+            views: {
+                'b@': {
+                    template: '<div>View B</div>',
+                    controller: 'BCtrl'
+                }
+            },
+            ncyBreadcrumb: {
+                label: 'State {{tripleB}}'
+            }
+        });
```
Comment (ncuillery) [inline test/mock/test-modules.js]: I think we could enrich this configuration with real multiple views to demonstrate it fixes #62 too, but I can add it later.

**Comment by ncuillery** [comment]
Review done :+1:

Please can you amend the commit and changing the title to `feat: add the scope-based ncyBreadcrumbIgnore flag` and add `Closes #42 #62` at the end of the commit message after a blank line.

Thank you for all of this :smile:

**Comment by kazinov** [comment]
will do today

**Comment by kazinov** [comment]
Hi, @ncuillery 
I changed the commit name and made some refactoring you asked for.

If you will update the documentation please consider mentioning the special case with inherited scopes.
Example [here](http://plnkr.co/edit/gSHA1sSmnUaDsNdWSFbQ?p=preview)

**Comment by ncuillery** [comment]
Landed: +1: the release will arrive soon.

I confirm it solves the #62 too !

**Pull request merged**